### PR TITLE
Use undefined object and jQuery trigger method

### PR DESCRIPTION
**Pull request by vsn4ik**
1. Use native focus method if possible
2. http://jsperf.com/is-undefined
3. Use `$.fn.trigger` for unification. Deprecate event shorthand methods (jQuery v4, https://github.com/jquery/jquery/issues/3214)

### CODE REVIEW

File: js/bootstrap-select.js [hunk 1]
```diff
@@ -558,16 +557,16 @@
...
+            icon = $this.data('icon') !== undefined ? '<span class="' + that.options.iconBase + ' ' + $this.data('icon') + '"></span> ' : '',
             isOptgroup = this.parentNode.tagName === 'OPTGROUP',
             isDisabled = this.disabled || (isOptgroup && this.parentNode.disabled);
 
         if (icon !== '' && isDisabled) {
           icon = '<span>' + icon + '</span>';
         }
 
-        if (that.options.hideDisabled && (isDisabled && !isOptgroup || this.parentNode.disabled && isOptgroup)) {
+        if (that.options.hideDisabled && isDisabled) {
```
Comment (caseyjhol) [inline js/bootstrap-select.js]: This line needs to be `that.options.hideDisabled && (isDisabled && !isOptgroup || this.parentNode.disabled && isOptgroup)`. With just `that.options.hideDisabled && isDisabled`, if an optgroup has a disabled option in it, the optgroup's header is hidden. I guess we could make a variable called something like `isOptgroupDisabled` and set it equal to `this.parentNode.disabled && isOptgroup`, and then set the if statement to `if (that.options.hideDisabled && (isDisabled && !isOptgroup || isOptgroupDisabled))`.

**Comment by vsn4ik** [comment]
@caseyjhol better looks now?

File: js/bootstrap-select.js [hunk 2]
```diff
@@ -598,19 +598,14 @@
             _li.push(generateLI(label, null, 'dropdown-header' + optGroupClass, optID));
           }
 
-          if (that.options.hideDisabled && isDisabled) {
-            liIndex--;
-            return;
-          }
-
```
Comment (caseyjhol) [inline js/bootstrap-select.js]: This needs to remain, otherwise disabled options within an optgroup stay visible.

Comment (vsn4ik) [inline js/bootstrap-select.js]: Done.

**Pull request closed without merging**